homeassistant/components/cisco_webex_teams/notify.py

```diff
@@ -0,0 +1,55 @@
+"""Cisco Webex Teams notify component."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.notify import (
+    PLATFORM_SCHEMA, BaseNotificationService, ATTR_TITLE)
+from homeassistant.const import (CONF_TOKEN)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['webexteamssdk==1.1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ROOM_ID = 'room_id'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_TOKEN): cv.string,
+    vol.Required(CONF_ROOM_ID): cv.string,
+})
+
+
+def get_service(hass, config, discovery_info=None):
+    """Get the CiscoWebexTeams notification service."""
+    return CiscoWebexTeamsNotificationService(
+        config[CONF_TOKEN],
+        config[CONF_ROOM_ID])
+
+
+class CiscoWebexTeamsNotificationService(BaseNotificationService):
+    """The Cisco Webex Teams Notification Service."""
+
+    def __init__(self, token, room):
+        """Initialize the service."""
+        from webexteamssdk import WebexTeamsAPI
+        self.room = room
+        self.client = WebexTeamsAPI(access_token=token)
+        # List all rooms which validates the token
+        # pylint:disable=expression-not-assigned
+        [room for room in self.client.rooms.list()]
```

Comment:
Create the client in `get_service` and check the token before creating the notification service instance. Pass the client to that instance.

Is it enough to call `client.rooms.list()` to check the token?

Reply:
> Is it enough to call client.rooms.list() to check the token?
 
Tried that but it doesn't through until you expand it:

```python
ipdb> self.client.rooms.list()
<GeneratorContainer list(self=<webexteamssdk.api.rooms.RoomsAPI object at 0x103d8f208>)>
ipdb>
```

```python
ipdb>  [room for room in self.client.rooms.list()]
*** webexteamssdk.exceptions.ApiError: [401] Unauthorized - The request requires a valid access token set in the Authorization request header.
ipdb>
```

Reply:
Maybe we don't need to iterate all the rooms? Try if `next(self.client.rooms.list())` is enough to trigger the exception.

Reply:
We should also catch the `ApiError` and then log error and `return None`.